tests/builder_test.h

```diff
@@ -0,0 +1,260 @@
+#ifndef BUILDER_TEST_H
```

Comment:
maybe call this file `test_builder.h` so it sorts with the other cpp test files.